c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -660,18 +660,20 @@ std::shared_ptr<void> getIpcDevPtr(std::string handle) {
   return sp;
 }
 
-cudaError_t
-Legacy::Alloc(void** refPtr, size_t nbytes, cudaStream_t stream) {
-  caching_allocator.malloc(refPtr, nbytes, stream);
-  return cudaSuccess;
+void* raw_alloc(size_t nbytes) {
+  int device;
+  C10_CUDA_CHECK(cudaGetDevice(&device));
```

Comment:
why not current_device from CUDAFunctions?

Reply:
No good reason. I see there are a bunch of other sites to fix too, so I'll do them all in one go in a follow up commit.